plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReloader.java

```diff
@@ -0,0 +1,329 @@
+
+    /**
+     * auto reload 1 model under the node id
+     *
+     * @param localNodeId node id
+     * @param modelId     model id
+     */
+    @VisibleForTesting
+    void autoReloadModelByNodeAndModelId(String localNodeId, String modelId) throws MLException {
+        String[] allNodeIds = nodeHelper.getAllNodeIds();
+        List<String> allNodeIdList = new ArrayList<>(List.of(allNodeIds));
+        if (!allNodeIdList.contains(localNodeId)) {
+            allNodeIdList.add(localNodeId);
+        }
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, allNodeIdList.toArray(new String[] {}), false, false);
+
+        client
+            .execute(
+                MLLoadModelAction.INSTANCE,
+                mlLoadModelRequest,
+                ActionListener
+                    .wrap(response -> log.info("the model {} is auto reloading under the node {} ", modelId, localNodeId), exception -> {
+                        log.error("fail to reload model " + modelId + " under the node " + localNodeId + "\nthe reason is: " + exception);
+                        throw new MLException(
+                            "fail to reload model " + modelId + " under the node " + localNodeId + "\nthe reason is: " + exception
+                        );
+                    })
+            );
+    }
+
+    /**
+     * query task index, and get the result of "task_type"="LOAD_MODEL" and "state"="COMPLETED" and
+     * "worker_node" match nodeId
+     *
+     * @param localNodeId one of query condition
+     */
+    @VisibleForTesting
+    void queryTask(String localNodeId, ActionListener<SearchResponse> searchResponseActionListener) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
+
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .must(QueryBuilders.matchPhraseQuery("task_type", "LOAD_MODEL"))
+            .must(QueryBuilders.matchPhraseQuery("worker_node", localNodeId))
+            .must(
+                QueryBuilders
+                    .boolQuery()
+                    .should(QueryBuilders.matchPhraseQuery("state", "COMPLETED"))
+                    .should(QueryBuilders.matchPhraseQuery("state", "COMPLETED_WITH_ERROR"))
+            );
+        searchSourceBuilder.query(queryBuilder);
+
+        SortBuilder<FieldSortBuilder> sortBuilderOrder = new FieldSortBuilder("create_time").order(SortOrder.DESC);
+        searchSourceBuilder.sort(sortBuilderOrder);
+
+        SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client, SearchAction.INSTANCE)
+            .setIndices(ML_TASK_INDEX)
+            .setSource(searchSourceBuilder);
+
+        searchRequestBuilder.execute(ActionListener.wrap(searchResponseActionListener::onResponse, exception -> {
+            log.error("index {} not found, the reason is {}", ML_TASK_INDEX, exception);
+            throw new IndexNotFoundException("index " + ML_TASK_INDEX + " not found");
```

Comment:
We can't confirm this is `IndexNotFoundException`, please throw a MLException instead, and please wrap the original exception into the MLException like this:`throw new MLException(exception)`

Reply:
ok, I have changed it~